src/environment.cpp

```diff
@@ -1 +1,17 @@
-#include "environment.hpp"
\ No newline at end of file
+#include "environment.hpp"
+#include "location.hpp"
+
+Environment::Environment(int width, int height) {
+    this->width = width;
+    this->height = height;
+    this->locations = std::vector<shared_ptr<Location>>(0);
+    for (int i = 0; i < width * height; i++) {
+        locations.push_back(shared_ptr<Location>(new Location()));
```

Comment:
Best to use make_shared to create shared ptr instances as it's exception safe. see https://docs.microsoft.com/en-us/cpp/cpp/how-to-create-and-use-shared-ptr-instances?view=msvc-170